cmd/root.go

```diff
@@ -134,11 +138,13 @@ func rootCmd(execs execs) (*cobra.Command, error) {
 			if compressionVar != "" {
 				compressionAlgo = compressionVar
 			}
-			if compressionAlgo != "" {
-				compressor, err = compression.GetCompressor(compressionAlgo)
-				if err != nil {
-					return fmt.Errorf("failure to get compression '%s': %v", compressionAlgo, err)
-				}
+			if compressionAlgo == "" {
```

Comment:
I don't understand the change here. Is this not essentially the same as before?

Reply:
When `--compression` is not set in the command line and the configuration file does not contain `compression`, you will get the following error:

```shell
time="2023-11-21T01:08:58Z" level=info msg="beginning dump 2023-11-21T01:08:58Z"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0xad3807]
goroutine 1 [running]:
github.com/databacker/mysql-backup/pkg/core.Dump({{0xc0001fb4a0, 0x1, 0x1}, 0x0, {0x0, 0x0, 0x0}, {{0x0, 0x0}, {0x0, ...}, ...}, ...})
/src/mysql-backup/pkg/core/dump.go:61 +0x1e7
github.com/databacker/mysql-backup/pkg/core.TimerDump({{0xc0001fb4a0, 0x1, 0x1}, 0x0, {0x0, 0x0, 0x0}, {{0x0, 0x0}, {0x0, ...}, ...}, ...}, ...)
/src/mysql-backup/pkg/core/dump.go:31 +0x13e
github.com/databacker/mysql-backup/cmd.dumpCmd.func2(0xc0000ff340?, {0xc657b8?, 0x1?, 0x1?})
/src/mysql-backup/cmd/dump.go:152 +0x7ed
github.com/spf13/cobra.(*Command).execute(0xc0000ff340, {0xc0001fa5d0, 0x1, 0x1})
/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:842 +0x67c
github.com/spf13/cobra.(*Command).ExecuteC(0xc0000ff080)
/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950 +0x39d
github.com/spf13/cobra.(*Command).Execute(...)
/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887
github.com/databacker/mysql-backup/cmd.Execute()
/src/mysql-backup/cmd/root.go:226 +0x68
main.main()
/src/mysql-backup/main.go:8 +0x17
```

The reason is actually that compression has no default value.

Reply:
> When --compression is not set in the command line

Then it should get the default:

```go
const (
	defaultCompression      = "gzip"
)
// ... lots of lines
	flags.String("compression", defaultCompression, "Compression to use. Supported are: `gzip`, `bzip2`")
```

Reply:
This will cause the same problem as the port being always overwritten as 3306.

Reply:
But that is the way it is now. Why is it causing a null pointer error. It is confusing me.

Reply:
We should use the same fix here as with the port.